chore(ajax cache): added ajax cache (#1181)
ing-bank/lion: packages/ajax/README.md

`;
```

## Ajax Cache

A caching library that uses `lion-web/ajax` and adds cache interceptors to provide caching for use in
frontend `services`.

> Technical documentation and decisions can be found in
> [./docs/technical-docs.md](./docs/technical-docs.md)

### Getting started

Consume the global `ajax` instance and add the interceptors to it, using a cache configuration
which is applied on application level. If a developer wants to add specifics to cache behavior
he or she has to provide a cache config per action (get etc) via `lionCacheOptions` field of local ajax config,
see examples below.

> **Note**: make sure to add the **interceptors** only **once**. This is usually
> done on app-level

```js
import {
  ajax,
  lionCacheRequestInterceptorFactory,
  lionCacheResponseInterceptorFactory,
} from '@lion-web/ajax.js';

const globalCacheOptions = {
  useCache: 'always',
  timeToLive: 1000 * 60 * 5, // 5 minutes
};

> const getCacheIdentifier = () => getActiveProfile().refreshToken;

Reviewer: This is probably very application specific. If we do show the profile, we shouldn't use the `refreshToken` because it will change each refresh. We want to use the profileId here.

Author: thanks @LarsDenBakker

`;
```

## Ajax Cache

A caching library that uses `lion-web/ajax` and adds cache interceptors to provide caching for use in
frontend `services`.

> Technical documentation and decisions can be found in
> [./docs/technical-docs.md](./docs/technical-docs.md)

### Getting started

Consume the global `ajax` instance and add the interceptors to it, using a cache configuration
which is applied on application level. If a developer wants to add specifics to cache behavior

> he or she has to provide a cache config per action (get etc) via `lionCacheOptions` field of local ajax config,

Reviewer 2: ```suggestion
they have to provide a cache config per action (`get`, `post`, etc.) via `lionCacheOptions` field of local ajax config,
```

ajax.interceptors.response.use(
  lionCacheResponseInterceptorFactory(getCacheIdentifier, cacheOptions),
);

class TodoService {
  constructor() {
    this.localAjaxConfig = {
      lionCacheOptions: {
        invalidateUrls: ['/api/todosbykeyword'], // default: []
        // invalidateUrlsRegex: RegExp, // see docs below for more info
        // requestIdentificationFn: (requestConfig) => { }, // see docs below for more info
      },
    };
  }

  /**
   * Returns all todos from cache if not older than 5 minutes
   */
  getTodos() {
    return ajax.get(`//api.${hostname}/api/todos`, this.localAjaxConfig);
  }

  /**
   *
   */
  getTodosByKeyword(keyword) {
    return ajax.get(`//api.${hostname}/api/todosbykeyword/${keyword}`, this.localAjaxConfig);
  }

  /**
   * Creates new todo and invalidates cache.
   * `getTodos` will NOT take the response from cache
   */
  saveTodo(todo) {
    return ajax.post(`//api.${hostname}/api/todos`, { data: todo }, this.localAjaxConfig);
  }
}
```

### Ajax cache Options

```js
const cacheOptions = {
  // `useCache`: determines wether or not to use the cache
  // can be string 'always' | 'never'
  // default: 'never'

>   useCache: 'always',

Author: )I think we can change it to boolean, it was done like this (in the beginning) for readability but, I agree, it does not really give here any benefit
